### DISCUSSION TITLE
Testing in turborepo

### DISCUSSION THREAD

**Discussion by andrevenancio** [vercel/turborepo#5590] [Help] [answered]
Hi, I'm using a similar setup to the one's found on the examples folder.

I have some typescript files and if those files have NO external dependency, then my tests run.
If one of them have a dependency to another module, what happens is that turborepo prioritises the `dist` folder content for that module and then the tests fail because we're mixing typescript and javascript. I can't figure out how to write tests on modules inside my package.

**Comment by anthonyshew** [comment, marked as answer]
It sounds like you're looking for:
```
{
  "pipeline": {
    "test": {
      "dependsOn": ["build"]
  }
}
```

Notice the lack of `^`, meaning that your package needs to `build` itself before it runs `test`.

I may be misunderstanding the question here, though. Please let me know!

**Reply by anthonyshew** [reply]
Going to mark this as answer as close since I haven't heard back!
